app/components/button_component.rb

```diff
@@ -1,10 +1,8 @@
 class ButtonComponent < BaseComponent
   attr_reader :action, :icon, :outline, :tag_options
 
-  DEFAULT_BUTTON_TYPE = :button
-
   def initialize(
-    action: ->(content, **tag_options) { button_tag(content, **tag_options) },
+    action: ->(**tag_options, &block) { button_tag(**tag_options, &block) },
```

Comment:
so now just to clarify, we've stopped doing the positional args at all, content *must* come from the block

It *might* be overkill but maybe we should wrap the block, so we can check to make sure it was yielded to? ex in the button component, just to save a step if somebody ever had to debug "why didn't my content show up?"

```
def content
  @was_called = true
  super
end

# or some other hook
def after_render
  raise "content block was not called!!! don't forget to yield or pass &block in action" if @was_called
  super
end
```

Reply:
> so now just to clarify, we've stopped doing the positional args at all, content _must_ come from the block

Correct 👍  Though, based on how ViewComponent works, this can also be done with `.with_content` and produce the same outcome.

> It _might_ be overkill but maybe we should wrap the block, so we can check to make sure it was yielded to? ex in the button component, just to save a step if somebody ever had to debug "why didn't my content show up?"

It's pretty easy to hook `before_render` (that's the actual method name) but a corresponding `after_render` doesn't exist, best I can tell. We could achieve the same effect by overriding the `render_in` method like what we did in #5475. To be honest, I'm not super-convinced it's necessary, or at least I can never get a comfortable gauge on _how much validation is too much validation_, and this feels pretty borderline to me.

Reply:
Sounds good, we can leave out validation and always come back if we get burned by it